### ISSUE TITLE
IsValidShopDomainAsync is broke.

### ISSUE THREAD

**Issue by Pokoto** [nozzlegear/ShopifySharp#331]
Hi,

I experienced issues with my App and discovered that since a couple of days, 
IsValidShopDomainAsync stopped working - always returns false.

At first sight, looks likely Shopify changed the backend behavior 
(maybe a redirect on /admin or is checking for other headers...)

Testing against the domain home (which requires to set  a user agent)
seems to do the trick.

Here is a quick fix.

 public static async Task<bool> IsValidShopDomainAsync(string url) {
      
            var uri = ShopifyService.BuildShopUri(url, false);

            using (var client = new HttpClient())
            {

                string UserAgent = "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0";
                var msg = new HttpRequestMessage(HttpMethod.Head, uri);
                msg.Headers.Add("User-Agent", UserAgent);

                using (msg)
                {
                    try
                    {
                        var response = await client.SendAsync(msg);

                        return response.Headers.Any(h => h.Key.Equals("X-ShopId", StringComparison.OrdinalIgnoreCase));
             
                    }
                    catch (HttpRequestException)
                    {
                        return false;
                    }
                }
            }
        }

Best,
Andrea

**Comment by nozzlegear** [comment]
Thanks for reporting this Andrea, I've been seeing the same thing off and on starting about a week ago. Sometimes it works and sometimes it doesn't. This is an undocumented "feature" that the package has been using, so unfortunately there's no warning when Shopify makes a change that might break it. I'll get it fixed!

**Comment by Mat-Kay** [comment]
I am also seeing this as of today. Fiddler is showing a chain of redirects, ending in a 404:

```
HEAD /admin 
	-> 303 See Other, Location: Location: https://myshop.myshopify.com/admin/auth/login
HEAD /admin/auth/login 
	-> 302 Found, Location: https://app.shopify.com/services/login/identity?destination_uuid=***&return_to=***&ui_locales=en&ux=shop
HEAD https://app.shopify.com/services/login/identity?destination_uuid=***&return_to=***&ui_locales=en&ux=shop 
	-> 404 Not Found
```

The first 2 responses include the `X-ShopId` header, but the final 404 does not.

**Comment by nozzlegear** [comment]
Oh I see, because they're redirecting to some kind of auth service now instead of using a login page on the shop domain. So probably best to just make a request to the root domain instead of `/admin`, and not follow redirects.

**Comment by nozzlegear** [comment]
Got a fix coming later this evening. Hang tight everyone!

**Comment by nozzlegear** [comment]
Fixed in 4.18.1
